src/Misc/Util.cpp

```diff
@@ -255,4 +255,34 @@ char *rtosc_splat(const char *path, std::set<std::string> v)
     return buf;
 }
 
+void expanddirname(std::string &dirname) {
+    if (dirname.empty())
+        return;
+
+    // if the directory name starts with a ~ and the $HOME variable is
+    // defined in the environment, replace ~ by the content of $HOME
+    if (dirname.at(0) == '~') {
+        char *home_dirname = getenv("HOME");
+        if (home_dirname != NULL) {
+            dirname = std::string(home_dirname) + dirname.substr(1);
+        }
+    }
+
+#ifdef ZYN_DATADIR
+    {
+        std::string var = "$ZYN_DATADIR";
+        size_t i;
+        while ((i = dirname.find(var)) != std::string::npos) {
```

Comment:
I'd say let's replace these three lines with:

```
size_t pos = dirname.find(var);
if(pos != std::string::npos) {
    dirname.replace(pos, var.length(), ZYN_DATADIR);
```

and then things look good to my eyes. I don't see any sense in handling multiple expansions of this variable nor the fun possibility of recursive expansions.